### PR TITLE
access log: support logging source resource metadata

### LINKED ISSUE #5098: Access logging: Add option to log originating Ingress/HTTPProxy with custom operator
**Please describe the problem you have**
Knowing the namespace/name of the Ingress/HTTPProxy that apply to a route can be useful when analyzing access logs to debug an issue or useful when sending logs to an aggregation service in a multi-tenant cluster.

We can add an access log operator specific to contour that users can add to the "text" or JSON access log types that will enable logging the name and namespace of the "originating" resource.
Here we mean "originating" resource to be the Ingress/HTTPProxy/etc. which has the route configuration that the request causing the access log is matched to.
For example if we have an Ingress in the `default` namespace named `foo` that specifies a hostname and prefix path match `example.com/api`,  any request to `example.com/api/*` will come with `default` and `foo` in the access log.

This is similar to the nginx feature `$namespace` and `$ingress_name` logging variables: https://kubernetes.github.io/ingress-nginx/user-guide/nginx-configuration/log-format/

Possible implementations:
- Add metadata to each route and use the access log operators to fetch that and log it
  - https://github.com/envoyproxy/go-control-plane/blob/5903a350ed1cc44813f0a239ef3195278ac9a55d/envoy/config/route/v3/route_components.pb.go#L790-L795
  - https://www.envoyproxy.io/docs/envoy/latest/configuration/observability/access_log/usage#config-access-log-format-dynamic-metadata
- Add an access log with a different filter for each vhost/route etc.
  - filter on hostname, path etc?
  - https://github.com/envoyproxy/go-control-plane/blob/5903a350ed1cc44813f0a239ef3195278ac9a55d/envoy/config/accesslog/v3/accesslog.pb.go#L182onnection_manager.pb.go#L490
  - https://github.com/envoyproxy/go-control-plane/blob/5903a350ed1cc44813f0a239ef3195278ac9a55d/envoy/config/accesslog/v3/accesslog.pb.go#L262-L281
  - could cause a large number of access logs/processing decisions figuring out which match writes the log

Some considerations/testing notes:
- Need to make sure includes work properly, logging for included routes should work
- should work for Ingress
- an equivalent to the nginx `$service_name` might be useful as well, if not already available via existing Envoy access log operators
- naming?
- etc.

### PR DESCRIPTION
**Pull request by skriss**
Closes #5098.

Note, I haven't enabled any aliases for these fields in the text/envoy logging format yet. I do think we should probably do that, but it applies more generally than just for these new items. Currently it degrades gracefully in that if you get the format of the arguments inside the `%METADATA(...)%` operator wrong in your config, the access log will just show a blank (`-`).

### CODE REVIEW

File: site/content/docs/main/guides/structured-logs.md [hunk 1]
```diff
@@ -1,86 +0,0 @@
----
```
Comment (skriss) [inline site/content/docs/main/guides/structured-logs.md:1]: This was replaced by `config/access-logging.md` awhile back so dropping from the guides section.

File: internal/envoy/v3/route.go [hunk 2]
```diff
@@ -86,6 +87,17 @@ func VirtualHostAndRoutes(vh *dag.VirtualHost, dagRoutes []*dag.Route, secure bo
 func buildRoute(dagRoute *dag.Route, vhostName string, secure bool) *envoy_route_v3.Route {
 	route := &envoy_route_v3.Route{
 		Match: RouteMatch(dagRoute),
+		Metadata: &envoy_core_v3.Metadata{
+			FilterMetadata: map[string]*structpb.Struct{
+				"envoy.access_loggers.file": {
+					Fields: map[string]*structpb.Value{
+						"kind":      structpb.NewStringValue(dagRoute.Kind),
```
[the remark this answers is not in the document]
Comment (skriss) [reply internal/envoy/v3/route.go]: done https://github.com/projectcontour/contour/pull/5534/commits/ccecfc0714b5c10840b2102f96c0cc13a6b0ecd9

**Comment by skriss** [comment]
Started looking at updating tests here and as-is this is going to require updating hundreds of dag builder tests, feature tests, etc. by hand. Not sure I want to go down that path; one alternate option would be to make this opt-in behavior that is enabled in the actual Contour server but not for most DAG/feature tests, and then add some specific test cases that do enable it. Will play around some more..

**Review by sunjayBhatia — APPROVED** [review]
Tested it out with some various cases in my local cluster and it works great!

I was thinking about maybe in the future we could refactor more of the common dag element creation bits of the various processors with helper methods (similar to how we have `DAG.EnsureService`), in this case maybe change the `VirtualHost.AddRoute` method to do the common work and maybe have the `SetSourceMetadataOnRoutes` only in one place? I have mixed feelings on adding more logic to the dag itself so this would require a little more work so best to pull into its own idea/PR IMO

**Pull request merged**